test/signature.test.ts

```diff
@@ -0,0 +1,110 @@
+import * as nock from 'nock';
+
+import { RequestBuilder } from '../src/RequestBuilder';
+import { Holder, ISignature, Signature, SignatureState } from '../src';
+import { getFakeAlgoanServer, getOAuthServer } from './utils/fake-server.utils';
+import { folderSample } from './samples/folder';
+
+describe('Tests related to the Application class', () => {
+  const baseUrl: string = 'http://localhost:3000';
+  let signatureAPI: nock.Scope;
+  let requestBuilder: RequestBuilder;
+
+  beforeEach(() => {
+    getOAuthServer({
+      baseUrl,
+      isRefreshToken: false,
+      isUserPassword: false,
+      nbOfCalls: 1,
+      expiresIn: 500,
+      refreshExpiresIn: 2000,
+    });
+    requestBuilder = new RequestBuilder(baseUrl, {
+      clientId: 'a',
+      clientSecret: 's',
+    });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    nock.cleanAll();
+  });
+
+  describe('static createSignature()', () => {
+    beforeEach(() => {
+      signatureAPI = getFakeAlgoanServer({
+        baseUrl,
+        path: `/v1/folders/${folderSample.id}/signatures`,
+        response: {
+          state: 0,
+          legalDocumentIds: ['random_legal_id'],
+          id: '5f3663fc3ec5884afc8ed04f',
+          partnerId: 'SIGN_ID',
+          holder: 'APPLICANT',
+          createdAt: '2020-08-14T10:14:20.224+0000',
+        },
+        method: 'post',
+      });
+    });
+    it('should create a signature', async () => {
+      const signature: ISignature = await Signature.create(requestBuilder, folderSample.id, {
+        partnerId: 'SIGN_ID',
+        legalDocumentIds: ['random_legal_id'],
+        holder: Holder.APPLICANT,
+      });
+      expect(signatureAPI.isDone()).toBeTruthy();
+      expect(signature).toBeInstanceOf(Signature);
+      expect(signature.id).toEqual('5f3663fc3ec5884afc8ed04f');
+      expect(signature.legalDocumentIds).toEqual(['random_legal_id']);
+      expect(signature.partnerId).toEqual('SIGN_ID');
+      expect(signature.holder).toEqual('APPLICANT');
+      expect(signature.createdAt).toEqual(new Date('2020-08-14T10:14:20.224+0000'));
```

Comment:
What is the interest of these checks?

For me, these statements test fake server rather than our API. Am I wrong?